Source/WebGPU/WebGPU/BindGroup.mm

```diff
@@ -50,6 +50,35 @@ static bool textureViewIsPresent(const WGPUBindGroupEntry& entry)
     return entry.textureView;
 }
 
+#if HAVE(TIER2_ARGUMENT_BUFFERS)
```

Comment:
How can this be a preprocessor macro? It's hardware-dependent, not known at build time.

Also, why do we need any feature from tier 2 argument buffers at all? They're supposed to be able to work just fine with tier 1 only.

Reply:
Naming of `TIER2_ARGUMENT_BUFFERS` could be improved, it is more something along the lines of `HAVE(SUPPORTS_GPU_ADDRESS)`. I check the tier level inside the `#if` prior to building a tier 2 buffer.

We certainly don't need tier 2 argument buffers. I used it to simplify the implementation until we want to run this on pre-A13 iOS device (iPhone 8, iPhone X) or an Intel Mac with non-discrete GPU and I work out logic error in building the argument buffers from the GPUBindGroupDescriptors

Reply:
Most impacted in the pre-A13 category are iPads I think. I certainly think we need to support tier1 argument buffers, I plan to add support once I address remaining issues in the bind group implementation.